### PR TITLE
myReduce

### PR DESCRIPTION
**Pull request by Shapa291**

### CODE REVIEW

File: task1.js [hunk 1]
```diff
@@ -4,8 +4,24 @@ Array.prototype.myForEach = function (callback) {
     }
  }
  
+Array.prototype.myReduce = function (callback, initialValues) {
+    
+    let accamulator = ( initialValues === undefined ) ? this[0] : initialValues
+    const start = ( initialValues === undefined ) ? 1 : 0
+
+    for  (let i = start; i < this.length; i++) {
+       accamulator = callback (accamulator, this[i])
```
Comment (Satsuger) [inline task1.js]:
```suggestion
       accamulator = callback(accamulator, this[i], index, this) 
```

File: task1.js [hunk 2]
```diff
@@ -4,8 +4,24 @@ Array.prototype.myForEach = function (callback) {
     }
  }
  
+Array.prototype.myReduce = function (callback, initialValues) {
+    
+    let accamulator = ( initialValues === undefined ) ? this[0] : initialValues
+    const start = ( initialValues === undefined ) ? 1 : 0
```
Comment (Satsuger) [inline task1.js]: Original reduce hasn't such logic.

Let's repeat logic of original methods

File: task1.js [hunk 3]
```diff
@@ -4,8 +4,24 @@ Array.prototype.myForEach = function (callback) {
...
+    for  (let i = start; i < this.length; i++) {
+       accamulator = callback (accamulator, this[i]) 
+    }
+    return accamulator;
+}
+
  const arr = ["a", "b", "c"]
+
+ arr.myReduce(() => {
+     console.log();
```
Comment (Satsuger) [inline task1.js]: Please update the usage of the method to show a more obvious use case of this method.

File: task1.js (hunk 1 above)
Comment (Shapa291) [reply task1.js]: I agree with your remark

File: task1.js (hunk 3 above)
Comment (Shapa291) [reply task1.js]: Ooops, i forgot about this, yes of course i'll do some basic tests

**Comment by Satsuger** [comment]
I suppose it's ready to be merged

**Pull request merged**